autoimpute/imputations/dataframe/series_imputer.py

```diff
@@ -0,0 +1,153 @@
+        SeriesImputer() creates a valid class instance. The instance is
+        used to set up an imputer and perform checks on arguments.
+
+        Args:
+            k (int, optional): number of repeated fits and transformations to
+                apply to reach a stable impution. Sefault is 3.
+                Value must be greater than or equal to 1.
+            n (int, optional): number of imputations to perform. Default is 5.
+                Value must be greater than or equal to 1.
+            predictors (str, iter, dict, optional): defaults to all, i.e.
+                use all predictors. If all, every column will be used for
+                every class prediction. If a list, subset of columns used for
+                all predictions. If a dict, specify which columns to use as
+                predictors for each imputation. Columns not specified in dict
+                but present in `strategy` receive `all` other cols as preds.
+            imp_kwgs (dict, optional): keyword arguments for each imputer.
+                Default is None, which means default imputer created to match
+                specific strategy. imp_kwgs keys can be either columns or
+                strategies. If strategies, each column given that strategy is
+                instantiated with same arguments. When strategy is `default`,
+                imp_kwgs is ignored.
+            seed (int, optional): seed setting for reproducible results.
+                Defualt is None. No validation, but values should be integer.
+            return_list (bool, optional): return m as list or generator.
+                Default is False. m imputations returned as generator. More
+                memory efficient. return as list if return_list=True
+        """
+        self.k = k
+        super(SeriesImputer, self).__init__(n=n, strategy=strategy, predictors=predictors, imp_kwgs = imp_kwgs, seed=seed, visit=visit, return_list=return_list)
+
+    @property
+    def k(self):
+        """Property getter to return the value of the k property."""
+        return self._k
+
+    @k.setter
+    def k(self, k_):
+        """Validate the k property to ensure it's Type and Value.
+
+        Args:
+            k_ (int): k passed as arg to class instance.
+
+        Raises:
+            TypeError: k must be an integer.
+            ValueError: k must be greater than zero.
+        """
+
+        # deal with type first
+        if not isinstance(k_, int):
+            err = "k must be an integer specifying number of repeated fits and transformations in a series of imputations."
+            raise TypeError(err)
+
+        # then check the value is greater than zero
+        if k_ < 1:
+            err = "k > 0. Cannot perform fewer than 1 imputation."
+            raise ValueError(err)
+
+        # otherwise set the property value for k
+        self._k = k_
+
+    def iterate_imputation(self, X, imp):
```

Comment:
Can we rename this to `_iterate_imputation`? I add the prefix to indicate that this method is an internal helper method. Users of the class should not call this method directly.